Source/Core/Core/Core.cpp

```diff
@@ -146,7 +154,7 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	return s_cpu_thread.joinable() ? tls_is_cpu_thread : !s_is_started;
```

Comment:
This is delicate code, so I'd rather not mess with it too much here; just declaring, then undeclaring should be fine, with appropriate comments to explain what's happening.